image/base/Dockerfile

```diff
@@ -127,7 +127,22 @@ RUN rm -f /etc/service
 COPY etc/  /etc
 COPY sbin/ /sbin
 
+FROM discourse_dependencies AS discourse_slim
+ARG DISCOURSE_BRANCH=main
+
```

Comment:
How about we just drop the discourse/discourse clone from the slim image? I don't think it's serving any purpose there.

That way, we can avoid the need for the extra complexity of the discourse_dependencies image.

Reply:
AFAIK the only thing which uses base:slim is the discourse_test build (that was the original motivation for us introducing the slim image - to reduce the test image size for github actions).

So I think we can get away with removing the clone from slim, and updating discourse_test to make its own clone